GetJobber/atlantis
# build: Enable the use of npm pack

## Motivations

Found out that `npm pack` is a thing and it gives you a local version of what gets released in npm. Kinda like prereleases, but local.

## There are 2 ways you can pack things. 

### Recommended way

This is documented in the read me

```
npm run pack @jobber/components
```

Would run the bootstrap first to ensure you have all the changes before packing. This is good for the times you updated design or hooks so it gets packed up with components.

The issue however is it takes a while since it has to bootstrap first.

### Only when you know what you're doing

If you're just changing one of the packages, you can pack that directly without the bootstrap

```
npm pack -w @jobber/components
```

## Changes



### Added

- Add `prepack` and `postpack` scripts to add entry points when packing the package.

### Changed

- cleanup scripts in components package.json to be self documenting

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

- run `npm run pack -w @jobber/components`
  - It should generate a `.tgz` file
- install file in your project `npm i path/to/file/generated-file-name.tgz`

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

![Random photo of Atlantis](https://loremflickr.com/672/400/atlantis)
